synapseclient/core/credentials/credential_provider.py

```diff
@@ -188,6 +188,16 @@ def _get_auth_info(self, syn, user_login_args):
         return user_login_args.username, None, None, token
 
 
+class EnvironmentVariableCredentialsProvider(SynapseCredentialsProvider):
+    """
+    Retrieves the user's auth token from an environment variable
+    """
+    ENVIRONMENT_VAR_NAME = "SYNAPSE_ACCESS_TOKEN"
```

Comment:
Elsewhere id doc and config this is referred to as an "auth token" (since it could also work with an Oauth token if needed. Maybe we should have the environment variable name.